content/reference/chain-providers.adoc

```diff
@@ -8,16 +8,22 @@ Cognitect
 
 ifdef::env-github,env-browser[:outfilessuffix: .adoc]
 
-The building blocks of Pedestal are not limited to web
-applications. The link:interceptors[interceptor] concept is very
-general and can be used in message-processing or dataflow applications
-just as well as request/reply web services.
+Pedestal uses a chain of interceptors to process incoming HTTP requests and provide outgoing HTTP responses.
+However, the underlying code, the interceptors and the interceptor chain that executes them, is more general.
 
-A _chain provider_ invokes a queue of interceptors in response to some
-stimulus. The job of a chain provider is to accept incoming stimuli
-(requests, messages, events, commands, etc.) set up the
-link:context-map[context map] and execute a queue of interceptors via
-link:../api/pedestal.interceptor/io.pedestal.interceptor.chain.html#var-execute[`execute`].
+Interceptor chains can be used for a variety of operations, such as:
 
-For HTTP, the link:servlet-interceptor[servlet interceptor] is the
-chain provider.
+- Processing messages from a queue, such as Kafka or JMS
+- Performing a series of transformations on documents as part of a publishing service
+- Processing outgoing requests and incoming responses from a back-end service
+
+A _chain provider_ is responsible for accepting an incoming stimuli (such as an incoming HTTP request)
+and setting up a link:context-map [context map] and initial interceptors to process that stimuli.
+The service map is processed via the
```

Comment:
Is it worth underscoring that _service map_ here is distinct from the _context map_? I could see early readers missing the distinction and getting turned around.

Reply:
I think it's ok as it stands, discussion of service map is pretty localized to the parts of the documentation about standing up the service.  I think adding a sidebar or something here will negatively affect readability.